### ISSUE TITLE
Memory/RAM issue when writing MERSCOPE data (sopa read)

### ISSUE THREAD

**Issue by andrea-de-micheli** [gustaveroussy/sopa#32]
Hello,

My shell kills 'sopa read' although I do not seem to be exceeding my computer resources.

I also get this warning: 

```shell
UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
``` 

Trying to run on M2 Max Mac with 64 GB of RAM. Anyone with similar issues?

Thank you for your support.

**Comment by andrea-de-micheli** [comment]
Bonjour @quentinblampey,

Merci, I have MERSCOPE data and the 12 (x7 z layers) IF channels/panoramas are ~26GB each. Detected transcript file is ~9GB. My total merfish_output directory size is 2.1 TB.

sdata.write() is the step that crashes (zsh: killed python) and I then get the same warning. The directory tree of the zarr is created but no files are inside.

Thank you very much for your help!

**Comment by andrea-de-micheli** [comment]
Hello @quentinblampey,

Tried to run sopa read on a linux machine with 256 GB of ram and it crashes again. I can confirm it is RAM issue. How can I overcome this given the size of my merfish_output directory?

Thank you..

**Comment by quentinblampey** [comment]
Hello @andrea-de-micheli,

Okay so there is really something wrong with the RAM for the MERSCOPE reader...

Everything should run with a very low memory usage whatever the size of your input, so I have to fix this reader!

I'll let you know

**Comment by quentinblampey** [comment]
Hello again @andrea-de-micheli, I released a new version of Sopa (`v1.0.6`) with a new backend for MERSCOPE data using [`rioxarray`](https://github.com/corteva/rioxarray). It should use considerably less RAM, and it should ideally work without any problem on your Mac with 64GB of RAM.

The backend is still experimental, so for now you need to install rioxarray with `pip install rioxarray`. At some points, if this backend prove itself stable and much lighter in RAM usage, then it may become the default backend and I would then add it to the dependencies of Sopa.

For now, you have to specify this backend, i.e.:
- Via the API: `sdata = sopa.io.merscope("/path/to/data/directory", backend="rioxarray")`
- Via the CLI: use the `--kwargs` argument, for instance `--kwargs "{'backend': 'rioxarray'}"`
- Via the Snakemake pipeline: 
```yaml
read:
  technology: merscope
  kwargs:
    backend: rioxarray
```

If using the new backend, you should now see the log below when reading your MERSCOPE data. Else, check that you have indeed the new version `sopa==1.0.6`
```sh
[INFO] (sopa.io.transcriptomics) Using experimental rioxarray backend.
```
I hope it will fix your issue! Please let me know how it goes, it will help me improving Sopa :)

**Comment by andrea-de-micheli** [comment]
Hi @quentinblampey,

Thanks again for the swift response, highly appreciated! I've updated to v1.0.6 and I think something broke in sopa.io.merscope. I must confess that I haven't had the time to dive into your code. Irrespective of the backend I get the following no implementation error.

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/andrea/anaconda3/envs/sopa2/lib/python3.10/site-packages/sopa/io/transcriptomics.py", line 118, in merscope
    points[f"{dataset_id}_transcripts"] = _get_points(transcript_path)
  File "/Users/andrea/anaconda3/envs/sopa2/lib/python3.10/site-packages/sopa/io/transcriptomics.py", line 186, in _get_points
    transcripts = PointsModel.parse(
  File "/Users/andrea/anaconda3/envs/sopa2/lib/python3.10/functools.py", line 926, in _method
    return method.__get__(obj, cls)(*args, **kwargs)
  File "/Users/andrea/anaconda3/envs/sopa2/lib/python3.10/site-packages/spatialdata/models/models.py", line 507, in parse
    raise NotImplementedError()
**NotImplementedError**
```

Any ideas?

Clean Conda environment, python 3.10, rioxarray installed. Thanks again!

**Comment by quentinblampey** [comment]
Hi @andrea-de-micheli, I think this issue may be related to a version recent version of dask, as mentionned in [this issue](https://github.com/scverse/spatialdata-io/issues/117#issuecomment-1994313083). The error seems to be the same as yours, so maybe you can try his solution?
**TL;DR**: try installing `dask==2024.2.1` and then `sopa==1.0.6`

**Comment by andrea-de-micheli** [comment]
Thank you very much for your support. 
Issue resolved installing `dask==2024.2.1`
